model/src/main/java/de/danoeh/antennapod/model/feed/FeedPreferences.java

```diff
@@ -168,12 +191,30 @@ public void setFeedID(long feedID) {
         this.feedID = feedID;
     }
 
-    public boolean getAutoDownload() {
-        return autoDownload;
+    public boolean getAutoDownload(boolean globalDefault) {
+        switch (this.autoDownload) {
+            case ENABLED -> {
+                return true;
+            }
+            case DISABLED -> {
+                return false;
+            }
+            default -> {
+                return globalDefault;
+            }
+        }
+    }
+
+    public AutoDownloadSetting getAutoDownloadRaw() {
+        return this.autoDownload;
     }
 
     public void setAutoDownload(boolean autoDownload) {
-        this.autoDownload = autoDownload;
+        if (autoDownload) {
+            this.autoDownload = AutoDownloadSetting.ENABLED;
+        } else  {
+            this.autoDownload = AutoDownloadSetting.DISABLED;
+        }
```

Comment:
There should be a way to set it back to global, so this method should take an `AutoDownloadSetting`